pvlib/losses.py

```diff
@@ -85,3 +86,116 @@ def soiling_hsu(rainfall, cleaning_threshold, tilt, pm2_5, pm10,
+    manual_wash_dates : sequence or None, default None
+        A list or tuple of Python ``datetime.date`` when the panels were
+        manually cleaned. Note there is no grace period after a manual
+        cleaning, so soiling begins to build-up immediately after a manual
+        cleaning
+    initial_soiling : float, default 0
+        the initial fraction of soiling on the panels at time zero in the input
+
+    Returns
+    -------
+    soiling : timeseries
+        soiling build-up fraction
+
+    References
+    ----------
+    .. [1] "The Effect of Soiling on Large Grid-Connected Photovoltaic Systems
+    in California and the Southwest Region of the United States," Addriane
+    Kimber, et al., IEEE 4th World Conference on Photovoltaic Energy
+    Conference, 2006, :doi:`10.1109/WCPEC.2006.279690`
+    """
+    # convert grace_period to timedelata
+    grace_period = datetime.timedelta(days=grace_period)
+
+    # manual wash dates
+    if manual_wash_dates is None:
+        manual_wash_dates = []
+
+    # resample rainfall as days by summing intermediate times
+    rainfall = rainfall_timeseries.resample("D").sum()
+
+    # set indices to the end of the day
+    rainfall.index = rainfall.index + datetime.timedelta(hours=23)
+
+    # soiling
+    soiling = pd.Series(float('NaN'), index=rainfall_timeseries.index)
+
+    # set 1st timestep to initial soiling
+    soiling.iloc[0] = initial_soiling
+
+    # rainfall events that clean the panels
+    rain_events = rainfall > threshold
+
+    # loop over days
+    for today in rainfall.index:
+
+        # did rain exceed threshold?
+        rain_exceed_thresh = rainfall[today] > threshold
+
+        # if yes, then set soiling to zero
+        if rain_exceed_thresh:
+            soiling[today] = 0
+            initial_soiling = 0
+            continue
+
+        # start day of grace period
+        start_day = today - grace_period
+
+        # rainfall event during grace period?
+        rain_in_grace_period = any(rain_events[start_day:today])
+
+        # if rain exceeded threshold during grace period,
+        # assume ground is still damp, so no or v. low soiling
+        if rain_in_grace_period:
+            soiling[today] = 0
+            initial_soiling = 0
+            continue
+
+        # is this a manual wash date?
+        if today.date() in manual_wash_dates:
+            soiling[today] = 0
+            initial_soiling = 0
+            continue
+
+        # so, it didn't rain enough to clean, it hasn't rained enough recently,
+        # and we didn't manually clean panels, so soil them by adding daily
+        # soiling rate to soiling from previous day
+        total_soil = initial_soiling + soiling_rate
+
+        # check if soiling has reached the maximum
+        soiling[today] = (
+            max_soiling if (total_soil >= max_soiling) else total_soil)
+
+        initial_soiling = soiling[today]  # reset initial soiling
+
+    return soiling.interpolate()
```

Comment:
Since this interpolates between points spaced 24h apart, the soiling signal does not immediately drop straight to zero.  Is this intentional?  For example this shows a smooth drop to zero across jan 3rd:

```
rainfall = pd.Series(0, index=pd.date_range('2019-01-01', '2019-01-05', freq='h', closed='left'))
rainfall.loc['2019-01-03 10:00'] = 10
soiling_kimber(rainfall).plot()
```

Reply:
I'm okay with it like this, I figure it takes a day of rain to clean the panels, so in that example, it rained on Jan 3rd so by the end of the day, the panels should be clean
![Figure_1](https://user-images.githubusercontent.com/1385621/74419431-bb710680-4dfe-11ea-8971-8454e6204b95.png)
Are you okay with it like this? Any other suggestions?